Libraries/LibJS/Runtime/Value.cpp

```diff
@@ -673,4 +662,77 @@ bool abstract_eq(Interpreter& interpreter, Value lhs, Value rhs)
     return false;
 }
 
+TriState abstract_relation(Interpreter& interpreter, bool left_first, Value lhs, Value rhs)
+{
+    Value x_primitive = {};
+    Value y_primitive = {};
+
+    if (left_first) {
+        x_primitive = lhs.to_primitive(interpreter);
```

Comment:
I think we should update `Value::to_primitive()` to take a `PreferredType` and pass that on to `object::to_primitive()`and then be explicit and give it the number hint mentioned in the spec.

Reply:
Yeah, that would be useful. Would you like to do it as a part of this PR?

Reply:
sure, if you don't mind :)